packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -0,0 +1,635 @@
+      activeTickMarkColor: Color.lerp(a.activeTickMarkColor, b.activeTickMarkColor, t),
+      inactiveTickMarkColor: Color.lerp(a.inactiveTickMarkColor, b.inactiveTickMarkColor, t),
+      disabledActiveTickMarkColor:
+          Color.lerp(a.disabledActiveTickMarkColor, b.disabledActiveTickMarkColor, t),
+      disabledInactiveTickMarkColor:
+          Color.lerp(a.disabledInactiveTickMarkColor, b.disabledInactiveTickMarkColor, t),
+      thumbColor: Color.lerp(a.thumbColor, b.thumbColor, t),
+      disabledThumbColor: Color.lerp(a.disabledThumbColor, b.disabledThumbColor, t),
+      overlayColor: Color.lerp(a.overlayColor, b.overlayColor, t),
+      valueIndicatorColor: Color.lerp(a.valueIndicatorColor, b.valueIndicatorColor, t),
+      thumbShape: t < 0.5 ? a.thumbShape : b.thumbShape,
+      valueIndicatorShape: t < 0.5 ? a.valueIndicatorShape : b.valueIndicatorShape,
+      showValueIndicator: t < 0.5 ? a.showValueIndicator : b.showValueIndicator,
+    );
+  }
+
+  @override
+  String toString() =>
+      '$runtimeType(activeRail: $activeRailColor, thumbShape: ${thumbShape.runtimeType} etc...)';
+}
+
+/// Base class for slider thumb and value indicator shapes.
+///
+/// Create a subclass of this if you would like a custom slider thumb or
+/// value indicator shape.
+///
+/// See also:
+///
+///  * [DefaultSliderThumbShape] for a simple example of a thumb shape.
+///  * [DefaultSliderValueIndicatorShape], for a complex example of a value
+///    indicator shape.
+abstract class SliderComponentShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const SliderComponentShape();
+
+  /// Returns the preferred size of the shape, based on the given conditions.
+  Size getPreferredSize(bool isEnabled, bool isDiscrete);
+
+  /// Paints the shape, taking into account the state passed to it.
+  ///
+  /// [activationAnimation] is an animation triggered when the user beings
+  /// to interact with the slider. It reverses when the user stops interacting
+  /// with the slider.
+  /// [enableAnimation] is an animation triggered when the [Slider] is enabled,
+  /// and it reverses when the slider is disabled.
+  /// If [labelPainter] is non-null, then [labelPainter.paint] should be
+  /// called with the location that the label should appear. If the labelPainter
+  /// passed is null, then no label was supplied to the [Slider].
+  /// [value] is the current parametric value (from 0.0 to 1.0) of the slider.
+  void paint(
+    PaintingContext context,
+    bool isDiscrete,
+    Offset thumbCenter,
+    Animation<double> activationAnimation,
+    Animation<double> enableAnimation,
+    TextPainter labelPainter,
+    SliderThemeData sliderTheme,
+    TextDirection textDirection,
+    double textScaleFactor,
+    double value,
+  );
+}
+
+class DefaultSliderThumbShape extends SliderComponentShape {
```

Comment:
Probably shouldn't call this "Default" per https://github.com/flutter/flutter/pull/14741#discussion_r168654724

Reply:
Good point (and Ian too).  Changed to RoundSliderThumbShape.